### ISSUE TITLE
Log messages get duplicated when not using batching

### ISSUE THREAD

**Issue by yelvert** [SumoLogic/js-sumo-logger#71]
When `interval` is `0` (batching not used), if multiple log messages are send in rapid succession, then some of those messages get sent to sumo multiple times. This happens because of how the `pendingLogs` are being used in the code.

**Comment by billsaysthis** [comment]
@yelvert Which version of the library are you using? We had a similar issue a few months back and updated the code to correct things in v2.3.0.

**Comment by yelvert** [comment]
I'm currently using 2.3.1 and still having the issue.

**Comment by billsaysthis** [comment]
I'm thinking (hoping) https://github.com/SumoLogic/js-sumo-logger/pull/72 will help once it gets reviewed and merged.

**Comment by billsaysthis** [comment]
@yelvert The PR has been merged, can you test that it fixes your issue and if so please close this issue. Thanks!

**Comment by billsaysthis** [comment]
Closing as other trouble reports are confirming fix.

**Comment by yelvert** [comment]
I can confirm this issue is fixed for me as well.

**Comment by billsaysthis** [comment]
Thanks @yelvert

**Comment by c24w** [comment]
This still very much seems to be an issue.  We're logging quite aggressively (fire-and-forget) and seeing many duplicate logs coming through.

From a quick look, [`_postSuccess`](https://github.com/SumoLogic/js-sumo-logger/blob/664b1e16f3e2c13f8498dd472a4a09f98f9b7ba2/src/sumoLogger.js#L178) only dequeues pending logs when it gets a success response back.  That means if `log` is called before a previous `log` has a response, the second call to `log` sends both its log and the previous log.

Pseudo-flow:
```
log('log1');
// pendingLogs = ['log1'];
// 'log1' -> Sumo Logic

log('log2');
// pendingLogs = ['log1', 'log2'];
// 'log1' (duplicate), 'log2' -> Sumo Logic

// <- 'log1' success response
// pendingLogs = ['log2'];

// <- 'log1' (duplicate), 'log2' success response
// pendingLogs = [];
```

**Comment by billsaysthis** [comment]
I would definitely be open to a pull request here, unfortunately I'm slammed with getting a new Sumo feature out the door and won't have time for this in the next week or two.

**Comment by billsaysthis** [comment]
Should be fixed now with v2.6.0, thanks to @dcheng666666

**Comment by c24w** [comment]
From a cursory glance (on GitHub mobile) it looks like this fix might cause logs to be delayed or even lost.  For example, when no interval is configured, hence logs are only sent on each call to `log`:

```
log('log1');
// pendingLogs = ['log1'];
// 'log1' -> Sumo Logic

log('log2');
// pendingLogs = ['log1', 'log2'];
// logSending === true; nothing is sent

// <- 'log1' success response
// pendingLogs = ['log2'];
// logSending === false

// 'log2' only gets sent if there is another call to `log` at some point in the future
```
